SUBMISSION BY SIMON SAMUEL

0x-ximon opened the pull request.

0x-ximon committed:
chore: initialized next.js app
chore: initialized shadcn

0x-ximon committed:
chore: removed boilerplate page

0x-ximon committed:
feat: set up algorand connection
feat: added connect wallet button
feat: set up dark mode
feat: added app logo
chore: set up app layout
chore: added shadcn components

0x-ximon committed:
feat: account can now send transactions
feat: implemented application UI
chore: added more shadcn components

0x-ximon committed:
feat: can fetch asset data from pera public API
feat: added the scrollable assets lists
chore: removed localnet support

0x-ximon committed:
chore: updated README

ximon-x commented:
The application is currently deployed at [https://ximon-atom-swap.vercel.app/](https://ximon-atom-swap.vercel.app/).

0x-ximon committed:
feat: added atomic swap functionality
fix: resolved hidden donate button
feat: added Arc3 compatible ASA
chore: updated README

0x-ximon commented:
<h1 align="center">
 Pera Challenge Solution
<h1>
<p align="center">
  <img src="https://github.com/user-attachments/assets/53816ade-10d3-4756-9a08-2b1fe98361d0" />
<p>

## About Pera

Welcome to the **Pera Challenge**! Pera is the leading gateway for Algorand projects, providing essential tools and libraries for developers. This challenge simulates real-world scenarios in decentralized application (dApp) development using [Pera Connect](https://github.com/perawallet/connect) —a library designed to simplify wallet integration and blockchain interactions.

## 🥅 Objective

Build a simple web application using your favorite web framework (React, Vue, or any other) that:

- Connects to a user's Algorand address using Pera Connect.
- Handles transactions and displays verified assets.
- Provides a seamless and user-friendly experience.

## ✅ Completed Challenges

The app is currently deployed at [https://ximon-atom-swap.vercel.app/](https://ximon-atom-swap.vercel.app/).

- [x] Assets Verified List by Pera Public API
- [x] Multiple Account Support
- [x] Network Compatibility
- [x] Proper Error Handling
- [x] Asset Opt-In Functionality
- [x] Donation Feature
- [x] Atomic Swap

## 📑 Resources

- [Pera Developer Docs](https://docs.perawallet.app)
- [Pera Connect Documentation](https://github.com/perawallet/connect)
- [Pera Public API Documentation](https://docs.perawallet.app/references/public-api)
- [Algorand Developer Portal](https://developer.algorand.org/)
- [Algorand JavaScript SDK](https://github.com/algorand/js-algorand-sdk)

---

0x-ximon committed:
Update README.md
